### PR TITLE
Add TO Golang CRConfig generation

### PR DESCRIPTION
**Pull request by rob05c**
This should be ready to be merged.

It diffs identical with the Perl CRConfig generation.

It has tests, via sqlmock. I tried to add 'API tests' (and agree those are better), but the API test framework doesn't currently support Servers or DeliveryServices, and it looks like it won't be easy to add that support.

It doesn't change the API or GUI, so docs don't need changed.

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1029/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1030/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1031/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1252/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1257/
Test FAILed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1258/
Test PASSed.

File: traffic_ops/traffic_ops_golang/crconfig/config.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
+package crconfig
```
Comment (dangogh) [inline traffic_ops/traffic_ops_golang/crconfig/config.go]: missing Apache license header on several of the files here...

Comment (rob05c) [reply traffic_ops/traffic_ops_golang/crconfig/config.go]: Fixed

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1260/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1261/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1278/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1286/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1292/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1293/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1294/
Test PASSed.

**Comment by rob05c** [comment]
This should be ready to be merged.

It diffs identical with the Perl CRConfig generation.

It has tests, via sqlmock. I tried to add 'API tests' (and agree those are better), but the API test framework doesn't currently support Servers or DeliveryServices, and it looks like it won't be easy to add that support.

It doesn't change the API or GUI, so docs don't need changed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1322/
Test PASSed.

**Comment by dangogh** [comment]
I get compile errors:   missing a file,  maybe?
```
$ go build
# github.com/apache/incubator-trafficcontrol/traffic_ops/traffic_ops_golang/crconfig
crconfig/handler.go:100: undefined: GetSnapshot
crconfig/handler.go:161: undefined: Snapshot
crconfig/handler.go:204: undefined: Snapshot
```

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1340/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1477/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR/1478/
Test PASSed.

**Pull request merged**